Removing setup.py/requirements.txt check on cloud serverless docker deployments

bmrickman opened the pull request:
There's no reason to check for setup.py on a docker deployment.  The docker image should have everything needed to run the codeserver, regardless of whether or not a setup.py or requirements.txt exists.

## Summary & Motivation
Motivation:
I am developing a monorepo for my companies data engineering platform.  For simplicity, I'm currently using editable installs to manage dependencies between projects.  This works great for local development, and the simplest way for me to deploy is to package the whole repo in one docker image.  I have no need for a setup.py or requirements.txt at the base of my dockerfile.  I believe the inclusion of the setup check for docker deployments was accidental.  
Summary:
I just removed the call to _check_source_directory from the docker serverless deploy command.  I also removed the function entirely and put it directly into the deploy_python_executable_command function as it seemed like an unnecessary function call now that its only called in one place.  I also double checked that the function is not called outside of the file to make sure I don't cause regressions in other parts of the code.  

## How I Tested These Changes
I built my image, deployed it to ECR, and then deployed it to my dagster cloud deployment using the dagster-cloud CLI with my change.  I was able to see my asset graph in the UI and kick off my runs from the UI.

bmrickman committed:
Removing setup.py check on docker deployments

There's no reason to check for setup.py on a docker deployment.  The docker image should have everything needed to run the codeserver, regardless of whether or not a setup.py or requirements.txt exists.

bmrickman committed:
Merge branch 'dagster-io:master' into cloud_serverless_deploy_remove_setup_dot_py_check

clairelin135 reviewed (commented):
When a base image isn't provided, this deploy function falls back to using the standard python docker image. In this case I think it's still helpful to check the source directory for the python dependencies.

What do you think about just skipping the `_check_source_directory` check when a base image is provided?

bmrickman committed:
Update __init__.py

Setting _check_source_directory to run on docker deploy IF a base_image is not specified, in which case we presume the caller wants the cli to be responsible for building the base docker image.

bmrickman committed:
Update __init__.py

Some formatting cleanup

bmrickman commented:
I see, so you are supporting the case where the user wants the CLI to build the docker image for them?  That makes sense to me.  I have updated the PR in reponse to your request.

bmrickman committed:
Merge branch 'master' into cloud_serverless_deploy_remove_setup_dot_py_check
